[terra-slide-panel] Added ariaLabel prop to slide panel container

avinashg1994 opened the pull request:
### Summary
Added ariaLabel prop to the slide panel container with a default value as "Details Panel".
Also corrected the Doc test page to show default focus behaviour.

Resolves #889

avinashg1994 commented on packages/terra-slide-panel/src/SlidePanel.jsx:
```diff
@@ -53,6 +58,7 @@ const propTypes = {
 };
 
 const defaultProps = {
+  ariaLabel: 'Details Panel',
```
Do we require translations for the default value of `aria-label`?

neilpfeiffer replied:
I don't think it should have a default prop value -- as the generic slide-panel containter can be used for a wide variety of uses well beyond the common Details Panel, include some layouts using the child panel as the "primary"/navigation panel content that launches pre-opened with the `panelBehavior=squish` prop value. 

I would prefer to leave the new `ariaLabel` prop with no default value - force consumers to add it if their situation needs it for accessibity and label it correctly for their usage.

neilpfeiffer commented on packages/terra-slide-panel/src/SlidePanel.jsx:
```diff
@@ -11,6 +11,11 @@ const SlidePanelPositions = {
+  /**
+   * String that labels the Details Panel for screen readers.
+   */
+  ariaLabel: PropTypes.string,
```
prop name is too generic - not sure as a consumer whether this applies to the `main` content area or the `panel` content area. 

Would suggest changing this to `panelAriaLabel`, and if needed - also add a companion `mainAriaLabel`.

emilyrohrbough replied:
IMO this should be marked as required in the next MVB of slide-panel since this is required for accessibility. Should we log an issue to make this change on the next MVB?

ShettyAkarsh replied:
Yes i agree. will go ahead and log an issue.

avinashg1994 replied:
Issue link to update the props to be **Required** in the next MVB : https://github.com/cerner/terra-framework/issues/902

avinashg1994 commented:
Updated the following:

- Removed the default value
- Added `panelAriaLabel` and `mainAriaLabel` for the "Panel Content Area" and "Main Content area" respectively.
- Updated Jest tests and snapshots.
